lib/uring/uring.mli

```diff
@@ -91,42 +97,148 @@ module type FLAGS = sig
 module Open_flags : sig
   include FLAGS
 
-  val empty : t
   val append : t
+  (* TODO: its not clear what the behaviour of io_uring is with
+     append with respect to lseek. See https://github.com/axboe/liburing/issues/32.
+     Should we be exposing O_APPEND with io_uring at all? *)
+
   val cloexec : t
+  (** [cloexec] enables the close-on-exec flag for the new fd. *)
+
   val creat : t
+  (** [create] implies that if the pathname does not exist, it is
+      created as a regular file. *)
+
   val direct : t
+  (** [direct] disables the kernel buffer cache and performs IO
+      directly to and from the userspace buffers. *)
+
   val directory : t
+  (** [directory] causes the open operation to fail if the target
+      is not a directory. *)
+
   val dsync : t
+  (** [dsync] ensures that write operations on the file complete
+      according to the requirements of synchronised IO data integrity
+      completion. *) (* TODO bind fdatasync *)
+
   val excl : t
+  (** [excl] is used alongside {!creat} to ensure that the file
+      is created as a result of the {!openat2} call, and otherwise
+      fails with a {!Unix.EEXIST} exception. The only exception where
+      [excl] can be used without {!creat} is when attempting to open
+      block devices. If the block device is otherwise mounted, then
+      the open will fail with {!Unix.EBUSY}. *)
+
   val largefile : t
+  (* TODO this can be hidden, since we should always set _FILE_OFFSET_BITS=1
+     (and liburing already does that *)
+
   val noatime : t
+  (** [noatime] signals that the file access time should not be updated
+      when the file is read from. See {!Statx.atime_nsec}. *)
+
   val noctty : t
+  (** [noctty] ensures that if the path refers to a tty, it will not
+      be assigned as the controlling terminal even if one is not present. *)
+
   val nofollow : t
+  (** [nofollow] will cause the open to fail with {!Unix.ELOOP} if the
+      basename of the path is a symbolic link. *)
+
   val nonblock : t
+  (** [nonblock] will open the file in non-blocking mode. *)
+
   val path : t
+  (** [path] will obtain a fd that can only be used to either indicate
+      a location in a filesystem tree, or perform operations at the fd
+      level. The file is not opened, and so any IO operations on the file
+      will fail. [path] is only used with {!cloexec}, {!directory} and
+      {!nofollow}, and any other flags will be ignored. *)
+
   val sync : t
+  (** [sync] ensures that write operations on the file complete
+      according to the requirements of synchronised IO file integrity
+      completion. *) (* TODO bind fsync *)
+
   val tmpfile : t
+  (** [tmpfile] creates an anonymous temporary regular file. The pathname
+      must be a directory, within which an unnamed inode will be created.
+      If [tmpfile] is specified without {!excl}, then a subsequent
+      linkat call can move it permanently into the filesystem. *)
+  (* TODO bind symlinkat and linkat *)
```

Comment:
I also can't find any hint of readlinkat support in io_uring upstream, for some reason.